### PR TITLE
Added thermostat feature for heaters

### PR DESCRIPTION
**Pull request by joshuaswilcox**
Add Wall heater

### CODE REVIEW

**Comment by zyonse** [comment]
Thanks for the pull request, it looks like the changes you made to platform.ts should also fix #27

**Comment by joshuaswilcox** [comment]
> Thanks for the pull request, it looks like the changes you made to platform.ts should also fix #27

Yeah I believe they should. It seems that there is just a different json shape for some of their products

File: src/platform.ts [hunk 1]
```diff
@@ -128,7 +129,18 @@ export class DreoPlatform implements DynamicPlatformPlugin {
 
         // create the accessory handler for the restored accessory
         // this is imported from `platformAccessory.ts`
-        new FanAccessory(this, existingAccessory, state, ws);
+        switch (device.seriesName) {
+          case 'fan':
```
Comment (zyonse) [inline src/platform.ts:133]: Does Dreo actually use the name 'fan' for any devices in their API? Mine is named 'Tower Fan' which causes the plugin to skip over it and print 'error, unknown device type'

**Review by zyonse — CHANGES_REQUESTED** [review]

File: src/platform.ts [hunk 2]
```diff
@@ -128,7 +129,19 @@ export class DreoPlatform implements DynamicPlatformPlugin {
 
         // create the accessory handler for the restored accessory
         // this is imported from `platformAccessory.ts`
-        new FanAccessory(this, existingAccessory, state, ws);
+        switch (device.productName) {
+          case 'Tower Fan':
+          case 'Air Circulator':
```
Comment (joshuaswilcox) [inline src/platform.ts:134]: Oh much better!

**Review by zyonse — CHANGES_REQUESTED** [review]

**Comment by zyonse** [comment]
Can you post a screenshot of the HomeKit heater controls on your version? Does it still show a "cool" mode?

**Comment by joshuaswilcox** [comment]
> Can you post a screenshot of the HomeKit heater controls on your version? Does it still show a "cool" mode?

Yeah it still has the "cool" options, i couldnt figure out what to do about that, it seems that the only way to get the heat controls was with at "thermostat"  service but that also means it assumes it can cool.
Screenshot 2023-12-04 at 4 51 56 PM

**Comment by jbyerline** [comment]
I am also interested in this feature. I have a dreo 1500w floor space heater I would like to make HomeKit compatible. I'm unsure if these changes would work for other models but if it gets merged I can try it out once I have the heater in hand.

**Comment by zyonse** [comment]
Something like this might work: https://github.com/zyonse/homebridge-pentair-screenlogic/commit/a2f5f79cc79f8b321b87c5a34d0e86854df9fd5d

**Comment by mja00** [comment]
Any update on this? I've added a new heater to my Dreo account and now the plugin crashes lol

**Comment by zyonse** [comment]
Closing this due to inactivity. Feel free to open a new pull request when the requested changes are implemented

**Pull request closed without merging**